qiskit_experiments/library/characterization/t1.py

```diff
@@ -104,36 +94,15 @@ def _set_backend(self, backend: Backend):
                 timing_constraints=timing_constraints, scheduling_method=scheduling_method
             )
 
-        # Set conversion factor
-        if self.experiment_options.unit == "dt":
-            try:
-                dt_factor = getattr(self.backend.configuration(), "dt")
-                conversion_factor = dt_factor
-            except AttributeError as no_dt:
-                raise AttributeError("Dt parameter is missing in backend configuration") from no_dt
-        elif self.experiment_options.unit != "s":
-            conversion_factor = apply_prefix(1, self.experiment_options.unit)
-        else:
-            conversion_factor = 1
-        self.set_experiment_options(conversion_factor=conversion_factor)
-
     def circuits(self) -> List[QuantumCircuit]:
         """
         Return a list of experiment circuits
 
         Returns:
             The experiment circuits
-
-        Raises:
-            ValueError: When conversion factor is not set.
         """
-        prefactor = self.experiment_options.conversion_factor
-
-        if prefactor is None:
-            raise ValueError("Conversion factor is not set.")
-
         circuits = []
-        for delay in prefactor * np.asarray(self.experiment_options.delays, dtype=float):
+        for delay in np.asarray(self.experiment_options.delays, dtype=float):
```

Comment:
Can we explicitly convert delay into dt here? Because delay in SI is converted into dt in the transpiler and rounded to integer. So actual delay value can differ from x value in analysis by this rounded value. This effect becomes visible when T1 value is very short (really edge case though).

The code is something like

```python
delay_dt = int(np.round(delay / dt_factor))
...

circ.delay(delay_dt, 0, "dt")
...

    "xval": delay_dt * dt_factor,
```

same for T2

Reply:
Not sure. On the one hand, I understand your reason. On the other hand:
1. I was hoping to completely get rid of anything that has units and factors in it. 
2. Users may not understand why the delays they entered have shifted.

Reply:
I agree mismatch due to rounding error is negligible and probably this is overkill. In our device, the error of actual delay and fit xval is less than 0.22 ns which is way smaller than T1 decay of fit.

Reply:
Done, but in way that's a bit different from the one suggested here, please check

Reply:
It looks okay to me, but @nkanazawa1989 should check it. I don't know what one can rely on when doing a transpilation. Could a delay gate ever be split into multiple delays or could another delay gate ever get inserted in the beginning of a circuit?

Reply:
I think this is too hacky and has several edge cases. For example, you only assumes single delay in the program but this is not true because measure align pass will add extra delay before measurement to align acquire instruction. In addition, T2Ramsey needs delay to insert artificial rotation and the rotation angle should be computed with correct delay amount. Of course calling transpile for computing xval is too much extra computation overhead. Why just cannot 
```python
delays_dt = np.round(options.delays / self.backend.configuration().dt).astype(int)

for delay_dt in delays_dt:
    delay_sec_rounded = delay_dt * self.backend.configuration().dt  # used for T2Ramsey and xval
    qc.delay(delay_dt, 0, "dt")
```
Indeed this is what is done during transpilation.

Reply:
If you are still skeptical, you can call this function from terra. This is what exactly called when you transpile delays with value in sec.
https://github.com/Qiskit/qiskit-terra/blob/f6ab3a98ca2c8c33842e1adbf5da6fbde78dc268/qiskit/circuit/duration.py#L23-L42